src/ussd_popi_rapidpro.js

```diff
@@ -927,37 +927,195 @@ go.app = function() {
             });
         });
 
-        self.add("state_opt_out", function(name) {
+        self.states.add("state_optout_menu", function(name) {
             return new MenuState(name, {
-                question: $("Do you want to stop getting MomConnect messages?"),
+                question: $("What would you like to do?"),
+                choices: [
+                    new Choice("state_opt_out_reason", $("Stop getting messages")),
+                    new Choice("state_stop_being_part", $("Stop being part of research")),
+                    new Choice("state_anonymous_data", $("Make my data anonymous")),
+                    new Choice("state_no_optout", $("Nothing. I still want to get messages"))
+                ],
+                error: $("Try again, don't understand."),
+            });
+        });
+
+        self.add("state_stop_being_part", function(name) {
+           return new MenuState(name, {
+                question: $("If you stop being part of the research, you'll keep getting MomConnect " +
+                            "messages, but they might look a little different."),
                 error: $(
                     "Sorry we don't recognise that reply. Please enter the number next to your " +
                     "answer."
                 ),
                 choices: [
-                    new Choice("state_opt_out_reason", $("Yes")),
-                    new Choice("state_no_optout", $("No"))
+                    new Choice("state_stop_being_part_optout", $("Ok, continue")),
+                    new Choice("state_optout_menu", $("Go back"))
                 ]
             });
         });
 
-        self.add("state_no_optout", function(name) {
+        self.add("state_anonymous_data", function(name) {
             return new MenuState(name, {
-                question: $(
-                    "Thanks! MomConnect will continue to send helpful messages and process " +
-                    "your personal info. What would you like to do?"
+                question: $("If you make your data anonymous, we'll delete your phone number, " +
+                            "and we won't be able to send you messages." ,
+                            " " ,
+                            "Do you want to continue?"),
+                error: $(
+                    "Sorry we don't recognise that reply. Please enter the number next to your " +
+                    "answer."
                 ),
+                choices: [
+                    new Choice("state_anonymous_data_optout", $("Yes")),
```

Comment:
As per the new comments in Miro, this should also ask for a reason. I'm happy if we change this in a new PR to keep the size of this one down

Reply:
Will add to new PR